client/components/chart/legend.jsx

```diff
@@ -1,32 +1,19 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { find } from 'lodash';
+import React, { PropTypes, PureComponent, Component } from 'react';
+import { find, noop } from 'lodash';
 
 /**
  * Module variables
  */
-// TODO: remove and extend from PureComponent after porting the component to ES6
-const PureRenderMixin = require( 'react-pure-render/mixin' );
 
-const LegendItem = React.createClass( {
-	displayName: 'ModuleChartLegendItem',
-
-	mixins: [ PureRenderMixin ],
-
-	propTypes: {
-		checked: React.PropTypes.bool.isRequired,
-		label: React.PropTypes.oneOfType( [ React.PropTypes.object, React.PropTypes.string ] ),
-		attr: React.PropTypes.string.isRequired,
-		changeHandler: React.PropTypes.func.isRequired
-	},
-
-	clickHandler: function() {
+class LegendItem extends PureComponent {
+	clickHandler = () => {
 		this.props.changeHandler( this.props.attr );
-	},
+	}
 
-	render: function() {
+	render() {
 		return (
 			<li className="chart__legend-option">
 				<label htmlFor="checkbox" className="chart__legend-label is-selectable" onClick={ this.clickHandler } >
```

Comment:
`htmlFor` should reference an input's ID, but there is no ID on the checkbox -- since it's nested inside the label anyway, this can be removed.

Additionally, I think you can safely move the `clickHandler` to the input's `onChange` (if you remove the broken ID, it will allow clicking on the label to change the input, so that onChange will fire when clicking the label too).